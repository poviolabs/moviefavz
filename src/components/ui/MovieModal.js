import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

import {
  Skeleton,
  Modal,
  Row,
  Col,
  Typography,
  Divider,
  Button,
  Tooltip,
} from 'antd';
import { HeartOutlined, HeartFilled, LinkOutlined } from '@ant-design/icons';

import MovieRating from './MovieRating';
import MovieImage from './MovieImage';

const { Title, Paragraph } = Typography;

const StyledPosterImage = styled(MovieImage)`
  object-fit: cover;
  width: 100%;
  height: auto;
  margin-bottom: 24px;
`;

const StyledSmallText = styled(Paragraph)`
  font-size: 0.85rem;
  color: ${({ theme }) => theme.colors.textSecondary};
`;

const MovieModal = ({
  visible,
  loading,
  movie,
  favorited,
  onClose,
  onFavoritesPress,
}) => {
  return (
    <Modal
      title={movie?.Title}
      visible={visible}
      onCancel={onClose}
      footer={null}
      width={720}
    >
      {loading ? (
        <Skeleton paragraph={{ rows: 10 }} active />
      ) : (
        movie && (
          <>
            <Row gutter={16}>
              <Col sm={{ span: 24 }} lg={{ span: 10 }}>
                <StyledPosterImage poster={movie.Poster} title={movie.Title} />
                <Tooltip
                  placement="top"
                  title={favorited ? 'Click to remove from favorites' : ''}
                >
                  <Button
                    type="primary"
                    block
                    icon={favorited ? <HeartFilled /> : <HeartOutlined />}
                    size="large"
                    onClick={() => onFavoritesPress(movie.imdbID)}
                  >
                    {favorited ? 'Favorited' : 'Add to Favorites'}
                  </Button>
                </Tooltip>
                <Button
                  icon={<LinkOutlined />}
                  block
                  size="large"
                  onClick={() =>
                    window.open(
                      `https://www.imdb.com/title/${movie.imdbID}/`,
                      '_blank'
                    )
                  }
                >
                  View on IMDB
                </Button>
                <Divider />
              </Col>
              <Col sm={{ span: 24 }} lg={{ span: 14 }}>
                <Title level={3}>
                  {movie.Title} ({movie.Year})
                </Title>
                <StyledSmallText>Awards: {movie.Awards}</StyledSmallText>
                <Paragraph>{movie.Plot}</Paragraph>
                <Divider />
                <Title level={4}>Genres</Title>
                <StyledSmallText>{movie.Genre}</StyledSmallText>
                <Divider />
                <Title level={4}>Cast</Title>
                <StyledSmallText>
                  <strong>Actors:</strong> {movie.Actors}
                </StyledSmallText>
                <StyledSmallText>
                  <strong>Director:</strong> {movie.Director}
                </StyledSmallText>
                <Divider />
                <Title level={4}>Ratings</Title>
                {movie.Ratings?.map(({ Source, Value }) => (
                  <MovieRating key={Source} source={Source} value={Value} />
                ))}
                <MovieRating
                  source="IMDB Rating"
                  value={`${movie.imdbRating} (${movie.imdbVotes} votes)`}
                />
              </Col>
            </Row>
          </>
        )
      )}
    </Modal>
  );
};

MovieModal.propTypes = {
  visible: PropTypes.bool,
  loading: PropTypes.bool,
  onClose: PropTypes.func,
  movie: PropTypes.object,
  favorited: PropTypes.bool,
  onFavoritesPress: PropTypes.func,
};

export default MovieModal;
