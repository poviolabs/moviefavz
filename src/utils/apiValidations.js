const validResponse = (status) => status && status >= 200 && status < 300;

export default { validResponse };
