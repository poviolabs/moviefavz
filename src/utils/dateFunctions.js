import dayjs from 'dayjs';

const getUnixDay = (timestamp = Date.now()) => {
  return dayjs(timestamp).startOf('date').valueOf();
};

const fillDaysToMatch = (num, data) => {
  if (!data.length) {
    return [];
  }
  const [lastDate] = data.reverse();

  return Array.from('x'.repeat(num - data.length))
    .map((_, i) => i + 1)
    .reverse()
    .map((i) => {
      return {
        value: 0,
        timestamp: dayjs(lastDate.timestamp).subtract(i, 'day').valueOf(),
      };
    });
};

export default { getUnixDay, fillDaysToMatch };
