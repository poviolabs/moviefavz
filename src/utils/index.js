export { default as arrayFunction } from './arrayFunctions';
export { default as apiValidations } from './apiValidations';
