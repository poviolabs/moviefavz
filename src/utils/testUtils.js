import React from 'react';
import { render } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';

import { ThemeProvider } from 'styled-components';
import { theme } from '../styles';

// eslint-disable-next-line react/prop-types
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <BrowserRouter>{children}</BrowserRouter>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
