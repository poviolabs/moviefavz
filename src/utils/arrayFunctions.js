import dateFunctions from './dateFunctions';

const groupArr = (data, size) => {
  const group = [];
  for (let i = 0, j = 0; i < data.length; i++) {
    if (i >= size && i % size === 0) {
      j++;
    }
    group[j] = group[j] || [];
    group[j].push(data[i]);
  }
  return group;
};

const groupByDay = (data, key = 'timestamp') => {
  const groupedByKey = data.reduce((obj, item) => {
    const timestamp = item[key];
    const day = dateFunctions.getUnixDay(timestamp);

    if (Object.prototype.hasOwnProperty.call(obj, day)) {
      obj[day].value += 1;
    } else {
      obj[day] = {
        value: 1,
        timestamp,
      };
    }
    return obj;
  }, {});
  return Object.values(groupedByKey);
};

export default { groupArr, groupByDay };
