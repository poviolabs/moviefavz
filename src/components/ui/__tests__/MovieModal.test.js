import React from 'react';
import { render } from '../../../utils/testUtils';

import MovieModal from '../MovieModal';

describe('<MovieModal />', () => {
  it('Renders the Component', () => {
    const props = {
      visible: true,
      loading: false,
      onClose: jest.fn(),
      movie: {},
      favorited: true,
      onFavoritesPress: jest.fn(),
    };
    const container = render(<MovieModal {...props} />);
    expect(container).toMatchSnapshot();
  });
});
