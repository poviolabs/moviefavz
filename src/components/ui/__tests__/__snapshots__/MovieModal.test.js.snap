// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieModal /> Renders the Component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: hidden; overflow-x: hidden; overflow-y: hidden;"
  >
    <div />
    <div>
      <div
        class="ant-modal-root"
      >
        <div
          class="ant-modal-mask fade-appear"
        />
        <div
          class="ant-modal-wrap "
          role="dialog"
          tabindex="-1"
        >
          <div
            class="ant-modal zoom-appear"
            role="document"
            style="width: 720px;"
          >
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden; outline: none;"
              tabindex="0"
            />
            <div
              class="ant-modal-content"
            >
              <button
                aria-label="Close"
                class="ant-modal-close"
                type="button"
              >
                <span
                  class="ant-modal-close-x"
                >
                  <span
                    aria-label="close"
                    class="anticon anticon-close ant-modal-close-icon"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="close"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M563.8 512l262.5-312.9c4.4-5.2.7-13.1-6.1-13.1h-79.8c-4.7 0-9.2 2.1-12.3 5.7L511.6 449.8 295.1 191.7c-3-3.6-7.5-5.7-12.3-5.7H203c-6.8 0-10.5 7.9-6.1 13.1L459.4 512 196.9 824.9A7.95 7.95 0 00203 838h79.8c4.7 0 9.2-2.1 12.3-5.7l216.5-258.1 216.5 258.1c3 3.6 7.5 5.7 12.3 5.7h79.8c6.8 0 10.5-7.9 6.1-13.1L563.8 512z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <div
                class="ant-modal-body"
              >
                <div
                  class="ant-row"
                  style="margin-left: -8px; margin-right: -8px;"
                >
                  <div
                    class="ant-col ant-col-sm-24 ant-col-lg-10"
                    style="padding-left: 8px; padding-right: 8px;"
                  >
                    <img
                      class="sc-AxirZ fjUsES"
                    />
                    <button
                      class="ant-btn ant-btn-primary ant-btn-lg ant-btn-block"
                      type="button"
                    >
                      <span
                        aria-label="heart"
                        class="anticon anticon-heart"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="heart"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M923 283.6a260.04 260.04 0 00-56.9-82.8 264.4 264.4 0 00-84-55.5A265.34 265.34 0 00679.7 125c-49.3 0-97.4 13.5-139.2 39-10 6.1-19.5 12.8-28.5 20.1-9-7.3-18.5-14-28.5-20.1-41.8-25.5-89.9-39-139.2-39-35.5 0-69.9 6.8-102.4 20.3-31.4 13-59.7 31.7-84 55.5a258.44 258.44 0 00-56.9 82.8c-13.9 32.3-21 66.6-21 101.9 0 33.3 6.8 68 20.3 103.3 11.3 29.5 27.5 60.1 48.2 91 32.8 48.9 77.9 99.9 133.9 151.6 92.8 85.7 184.7 144.9 188.6 147.3l23.7 15.2c10.5 6.7 24 6.7 34.5 0l23.7-15.2c3.9-2.5 95.7-61.6 188.6-147.3 56-51.7 101.1-102.7 133.9-151.6 20.7-30.9 37-61.5 48.2-91 13.5-35.3 20.3-70 20.3-103.3.1-35.3-7-69.6-20.9-101.9z"
                          />
                        </svg>
                      </span>
                      <span>
                        Favorited
                      </span>
                    </button>
                    <button
                      class="ant-btn ant-btn-lg ant-btn-block"
                      type="button"
                    >
                      <span
                        aria-label="link"
                        class="anticon anticon-link"
                        role="img"
                      >
                        <svg
                          aria-hidden="true"
                          class=""
                          data-icon="link"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M574 665.4a8.03 8.03 0 00-11.3 0L446.5 781.6c-53.8 53.8-144.6 59.5-204 0-59.5-59.5-53.8-150.2 0-204l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3l-39.8-39.8a8.03 8.03 0 00-11.3 0L191.4 526.5c-84.6 84.6-84.6 221.5 0 306s221.5 84.6 306 0l116.2-116.2c3.1-3.1 3.1-8.2 0-11.3L574 665.4zm258.6-474c-84.6-84.6-221.5-84.6-306 0L410.3 307.6a8.03 8.03 0 000 11.3l39.7 39.7c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c53.8-53.8 144.6-59.5 204 0 59.5 59.5 53.8 150.2 0 204L665.3 562.6a8.03 8.03 0 000 11.3l39.8 39.8c3.1 3.1 8.2 3.1 11.3 0l116.2-116.2c84.5-84.6 84.5-221.5 0-306.1zM610.1 372.3a8.03 8.03 0 00-11.3 0L372.3 598.7a8.03 8.03 0 000 11.3l39.6 39.6c3.1 3.1 8.2 3.1 11.3 0l226.4-226.4c3.1-3.1 3.1-8.2 0-11.3l-39.5-39.6z"
                          />
                        </svg>
                      </span>
                      <span>
                        View on IMDB
                      </span>
                    </button>
                    <div
                      class="ant-divider ant-divider-horizontal"
                      role="separator"
                    />
                  </div>
                  <div
                    class="ant-col ant-col-sm-24 ant-col-lg-14"
                    style="padding-left: 8px; padding-right: 8px;"
                  >
                    <h3
                      class="ant-typography"
                    >
                       (
                      )
                    </h3>
                    <div
                      class="ant-typography sc-AxiKw jseXPk"
                    >
                      Awards: 
                    </div>
                    <div
                      class="ant-typography"
                    >
                      
                    </div>
                    <div
                      class="ant-divider ant-divider-horizontal"
                      role="separator"
                    />
                    <h4
                      class="ant-typography"
                    >
                      Genres
                    </h4>
                    <div
                      class="ant-typography sc-AxiKw jseXPk"
                    >
                      
                    </div>
                    <div
                      class="ant-divider ant-divider-horizontal"
                      role="separator"
                    />
                    <h4
                      class="ant-typography"
                    >
                      Cast
                    </h4>
                    <div
                      class="ant-typography sc-AxiKw jseXPk"
                    >
                      <strong>
                        Actors:
                      </strong>
                       
                    </div>
                    <div
                      class="ant-typography sc-AxiKw jseXPk"
                    >
                      <strong>
                        Director:
                      </strong>
                       
                    </div>
                    <div
                      class="ant-divider ant-divider-horizontal"
                      role="separator"
                    />
                    <h4
                      class="ant-typography"
                    >
                      Ratings
                    </h4>
                    <div
                      class="ant-typography sc-AxjAm cSKDkf"
                    >
                      <strong>
                        IMDB Rating
                        :
                      </strong>
                       
                      undefined (undefined votes)
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              aria-hidden="true"
              style="width: 0px; height: 0px; overflow: hidden; outline: none;"
              tabindex="0"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
